### ISSUE TITLE
Framework not downloaded with Carthage

### ISSUE THREAD

**Issue by gemedet** [segmentio/analytics-ios#722]
I added `github "segmentio/analytics-ios"` to my Cartfile and ran `carthage update --platform ios`. 

The build output lists
```
Downloading analytics-ios.framework binary at "Version 3.6.7 *(24th August, 2017)*"
```
But the framework doesn't actually appear in `/Carthage/Build/iOS`. Is this a Carthage issue (using v0.26)?

**Comment by luizrath** [comment]
@gemedet , I've found a workaround for this issue. Update your Cartfile to include the commit hash of release 3.6.7. 

`github "segmentio/analytics-ios" "c908c4c"`

Then run `carthage update analytics-ios --platform iOS`

It will force Carthage to check out and build the framework.

**Comment by gemedet** [comment]
Thanks! That's good as a temporary fix.

**Comment by f2prateek** [comment]
@luizrath do you know why the fix works? is there something we can do in this library to fix the issue?

**Comment by neilco** [comment]
@f2prateek The reason why we see 
```
Downloading analytics-ios.framework binary at "Version 3.6.7 *(24th August, 2017)*"
``` 
But the framework doesn't actually appear in `/Carthage/Build/iOS` is because **Carthage is only downloading `Analytics.framework.dSYM.zip` and not `Analytics.framework.zip`**. 

Also, looking at the structure of `Analytics.framework.zip`, it's clear it wasn't created using **`carthage archive Analytics`** (see the [Carthage docs](https://github.com/Carthage/Carthage#archive-prebuilt-frameworks-into-one-zip-file)). If it had been, it would have the following structure:

```shell
Carthage
└── Build
    ├── iOS
    │   ├── 3ED8692C-5D05-3A41-837A-C78B8A9AFBBE.bcsymbolmap
    │   ├── Analytics.framework
    │   ├── Analytics.framework.dSYM
    └── tvOS
        ├── 98EAF541-4461-3FB3-A700-8E0290B97255.bcsymbolmap
        ├── Analytics.framework
        └── Analytics.framework.dSYM
```

Using `carthage archive Analytics` would negate the need for a separate zip file containing the debug symbols.

**Comment by neilco** [comment]
BTW, a better working around than [specifying an exact commit hash in the Cartfile](https://github.com/segmentio/analytics-ios/issues/722#issuecomment-336981178) is to use the `--no-use-binaries` Carthage flag which will build the framework from source.

**Comment by f2prateek** [comment]
Sorry for the delay here. I think you're right - while do have a `make archive` (which runs `carthage archive Analytics`) target (https://github.com/segmentio/analytics-ios/blob/dev/Makefile#L18-L19), it wasn't added to the releasing steps https://github.com/segmentio/analytics-ios/blob/dev/RELEASING.md.

I ran it manually now, and that structure looks ok to me. 
[Analytics.framework.zip](https://github.com/segmentio/analytics-ios/files/1500244/Analytics.framework.zip)

So it seems to me once I update the releasing instructions and re-run `make carthage archive` and upload the new zip file, it should fix the issue.

**Comment by f2prateek** [comment]
Tracking internally as well in JIRA — https://segment.atlassian.net/browse/LIB-125

**Comment by f2prateek** [comment]
This should be fixed in 3.6.8. https://github.com/segmentio/analytics-ios/releases/tag/3.6.8

The issue was that we have two types of binaries, one for Carthage and one for manual installation. Previosly we were uploading the manual one and assumed Carthage could use it. However, whe two have subtle differences. Going forwards we'll release them as two separate binaries.
